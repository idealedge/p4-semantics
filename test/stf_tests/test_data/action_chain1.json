{
  "program" : "action_chain1.p4",
  "__meta__" : {
    "version" : [2, 7],
    "compiler" : "https://github.com/p4lang/p4c"
  },
  "header_types" : [
    {
      "name" : "scalars_0",
      "id" : 0,
      "fields" : []
    },
    {
      "name" : "data_t",
      "id" : 1,
      "fields" : [
        ["f1", 32, false],
        ["f2", 32, false],
        ["h1", 16, false],
        ["b1", 8, false],
        ["b2", 8, false]
      ]
    },
    {
      "name" : "extra_t",
      "id" : 2,
      "fields" : [
        ["h", 16, false],
        ["b1", 8, false],
        ["b2", 8, false]
      ]
    },
    {
      "name" : "standard_metadata",
      "id" : 3,
      "fields" : [
        ["ingress_port", 9, false],
        ["egress_spec", 9, false],
        ["egress_port", 9, false],
        ["clone_spec", 32, false],
        ["instance_type", 32, false],
        ["drop", 1, false],
        ["recirculate_port", 16, false],
        ["packet_length", 32, false],
        ["enq_timestamp", 32, false],
        ["enq_qdepth", 19, false],
        ["deq_timedelta", 32, false],
        ["deq_qdepth", 19, false],
        ["ingress_global_timestamp", 48, false],
        ["lf_field_list", 32, false],
        ["mcast_grp", 16, false],
        ["resubmit_flag", 1, false],
        ["egress_rid", 16, false],
        ["checksum_error", 1, false],
        ["_padding", 4, false]
      ]
    }
  ],
  "headers" : [
    {
      "name" : "scalars",
      "id" : 0,
      "header_type" : "scalars_0",
      "metadata" : true,
      "pi_omit" : true
    },
    {
      "name" : "standard_metadata",
      "id" : 1,
      "header_type" : "standard_metadata",
      "metadata" : true,
      "pi_omit" : true
    },
    {
      "name" : "data",
      "id" : 2,
      "header_type" : "data_t",
      "metadata" : false,
      "pi_omit" : true
    },
    {
      "name" : "extra[0]",
      "id" : 3,
      "header_type" : "extra_t",
      "metadata" : false,
      "pi_omit" : true
    },
    {
      "name" : "extra[1]",
      "id" : 4,
      "header_type" : "extra_t",
      "metadata" : false,
      "pi_omit" : true
    },
    {
      "name" : "extra[2]",
      "id" : 5,
      "header_type" : "extra_t",
      "metadata" : false,
      "pi_omit" : true
    },
    {
      "name" : "extra[3]",
      "id" : 6,
      "header_type" : "extra_t",
      "metadata" : false,
      "pi_omit" : true
    }
  ],
  "header_stacks" : [
    {
      "name" : "extra",
      "id" : 0,
      "header_type" : "extra_t",
      "size" : 4,
      "header_ids" : [3, 4, 5, 6]
    }
  ],
  "header_union_types" : [],
  "header_unions" : [],
  "header_union_stacks" : [],
  "field_lists" : [],
  "errors" : [
    ["NoError", 1],
    ["PacketTooShort", 2],
    ["NoMatch", 3],
    ["StackOutOfBounds", 4],
    ["HeaderTooShort", 5],
    ["ParserTimeout", 6]
  ],
  "enums" : [],
  "parsers" : [
    {
      "name" : "parser",
      "id" : 0,
      "init_state" : "start",
      "parse_states" : [
        {
          "name" : "extra",
          "id" : 0,
          "parser_ops" : [
            {
              "parameters" : [
                {
                  "type" : "stack",
                  "value" : "extra"
                }
              ],
              "op" : "extract"
            }
          ],
          "transitions" : [
            {
              "value" : "0x80",
              "mask" : "0x80",
              "next_state" : "extra"
            },
            {
              "value" : "default",
              "mask" : null,
              "next_state" : null
            }
          ],
          "transition_key" : [
            {
              "type" : "stack_field",
              "value" : ["extra", "b2"]
            }
          ]
        },
        {
          "name" : "start",
          "id" : 1,
          "parser_ops" : [
            {
              "parameters" : [
                {
                  "type" : "regular",
                  "value" : "data"
                }
              ],
              "op" : "extract"
            }
          ],
          "transitions" : [
            {
              "value" : "default",
              "mask" : null,
              "next_state" : "extra"
            }
          ],
          "transition_key" : []
        }
      ]
    }
  ],
  "deparsers" : [
    {
      "name" : "deparser",
      "id" : 0,
      "order" : ["data", "extra[0]", "extra[1]", "extra[2]", "extra[3]"]
    }
  ],
  "meter_arrays" : [],
  "counter_arrays" : [],
  "register_arrays" : [],
  "calculations" : [],
  "learn_lists" : [],
  "actions" : [
    {
      "name" : "NoAction",
      "id" : 0,
      "runtime_data" : [],
      "primitives" : []
    },
    {
      "name" : "NoAction",
      "id" : 1,
      "runtime_data" : [],
      "primitives" : []
    },
    {
      "name" : "NoAction",
      "id" : 2,
      "runtime_data" : [],
      "primitives" : []
    },
    {
      "name" : "NoAction",
      "id" : 3,
      "runtime_data" : [],
      "primitives" : []
    },
    {
      "name" : "NoAction",
      "id" : 4,
      "runtime_data" : [],
      "primitives" : []
    },
    {
      "name" : "set0b1",
      "id" : 5,
      "runtime_data" : [
        {
          "name" : "val",
          "bitwidth" : 8
        }
      ],
      "primitives" : [
        {
          "op" : "assign",
          "parameters" : [
            {
              "type" : "field",
              "value" : ["extra[0]", "b1"]
            },
            {
              "type" : "runtime_data",
              "value" : 0
            }
          ],
          "source_info" : {
            "filename" : "action_chain1.p4",
            "line" : 56,
            "column" : 14,
            "source_fragment" : "val) { modify_field(extra[0].b1, val)"
          }
        }
      ]
    },
    {
      "name" : "act1",
      "id" : 6,
      "runtime_data" : [
        {
          "name" : "val",
          "bitwidth" : 8
        }
      ],
      "primitives" : [
        {
          "op" : "assign",
          "parameters" : [
            {
              "type" : "field",
              "value" : ["extra[0]", "b1"]
            },
            {
              "type" : "runtime_data",
              "value" : 0
            }
          ],
          "source_info" : {
            "filename" : "action_chain1.p4",
            "line" : 59,
            "column" : 12,
            "source_fragment" : "val) { modify_field(extra[0].b1, val)"
          }
        }
      ]
    },
    {
      "name" : "act2",
      "id" : 7,
      "runtime_data" : [
        {
          "name" : "val",
          "bitwidth" : 8
        }
      ],
      "primitives" : [
        {
          "op" : "assign",
          "parameters" : [
            {
              "type" : "field",
              "value" : ["extra[0]", "b1"]
            },
            {
              "type" : "runtime_data",
              "value" : 0
            }
          ],
          "source_info" : {
            "filename" : "action_chain1.p4",
            "line" : 60,
            "column" : 12,
            "source_fragment" : "val) { modify_field(extra[0].b1, val)"
          }
        }
      ]
    },
    {
      "name" : "act3",
      "id" : 8,
      "runtime_data" : [
        {
          "name" : "val",
          "bitwidth" : 8
        }
      ],
      "primitives" : [
        {
          "op" : "assign",
          "parameters" : [
            {
              "type" : "field",
              "value" : ["extra[0]", "b1"]
            },
            {
              "type" : "runtime_data",
              "value" : 0
            }
          ],
          "source_info" : {
            "filename" : "action_chain1.p4",
            "line" : 61,
            "column" : 12,
            "source_fragment" : "val) { modify_field(extra[0].b1, val)"
          }
        }
      ]
    },
    {
      "name" : "noop",
      "id" : 9,
      "runtime_data" : [],
      "primitives" : []
    },
    {
      "name" : "noop",
      "id" : 10,
      "runtime_data" : [],
      "primitives" : []
    },
    {
      "name" : "noop",
      "id" : 11,
      "runtime_data" : [],
      "primitives" : []
    },
    {
      "name" : "noop",
      "id" : 12,
      "runtime_data" : [],
      "primitives" : []
    },
    {
      "name" : "noop",
      "id" : 13,
      "runtime_data" : [],
      "primitives" : []
    },
    {
      "name" : "setb2",
      "id" : 14,
      "runtime_data" : [
        {
          "name" : "val",
          "bitwidth" : 8
        }
      ],
      "primitives" : [
        {
          "op" : "assign",
          "parameters" : [
            {
              "type" : "field",
              "value" : ["data", "b2"]
            },
            {
              "type" : "runtime_data",
              "value" : 0
            }
          ],
          "source_info" : {
            "filename" : "action_chain1.p4",
            "line" : 55,
            "column" : 13,
            "source_fragment" : "val) { modify_field(data.b2, val)"
          }
        }
      ]
    },
    {
      "name" : "set1b1",
      "id" : 15,
      "runtime_data" : [
        {
          "name" : "val",
          "bitwidth" : 8
        }
      ],
      "primitives" : [
        {
          "op" : "assign",
          "parameters" : [
            {
              "type" : "field",
              "value" : ["extra[1]", "b1"]
            },
            {
              "type" : "runtime_data",
              "value" : 0
            }
          ],
          "source_info" : {
            "filename" : "action_chain1.p4",
            "line" : 57,
            "column" : 14,
            "source_fragment" : "val) { modify_field(extra[1].b1, val)"
          }
        }
      ]
    },
    {
      "name" : "set2b2",
      "id" : 16,
      "runtime_data" : [
        {
          "name" : "val",
          "bitwidth" : 8
        }
      ],
      "primitives" : [
        {
          "op" : "assign",
          "parameters" : [
            {
              "type" : "field",
              "value" : ["extra[2]", "b2"]
            },
            {
              "type" : "runtime_data",
              "value" : 0
            }
          ],
          "source_info" : {
            "filename" : "action_chain1.p4",
            "line" : 58,
            "column" : 14,
            "source_fragment" : "val) { modify_field(extra[2].b2, val)"
          }
        }
      ]
    },
    {
      "name" : "setb1",
      "id" : 17,
      "runtime_data" : [
        {
          "name" : "port",
          "bitwidth" : 9
        },
        {
          "name" : "val",
          "bitwidth" : 8
        }
      ],
      "primitives" : [
        {
          "op" : "assign",
          "parameters" : [
            {
              "type" : "field",
              "value" : ["data", "b1"]
            },
            {
              "type" : "runtime_data",
              "value" : 1
            }
          ],
          "source_info" : {
            "filename" : "action_chain1.p4",
            "line" : 50,
            "column" : 19,
            "source_fragment" : "val) { ..."
          }
        },
        {
          "op" : "assign",
          "parameters" : [
            {
              "type" : "field",
              "value" : ["standard_metadata", "egress_spec"]
            },
            {
              "type" : "runtime_data",
              "value" : 0
            }
          ],
          "source_info" : {
            "filename" : "action_chain1.p4",
            "line" : 50,
            "column" : 13,
            "source_fragment" : "port, val) { ..."
          }
        }
      ]
    }
  ],
  "pipelines" : [
    {
      "name" : "ingress",
      "id" : 0,
      "init_table" : "test1",
      "tables" : [
        {
          "name" : "test1",
          "id" : 0,
          "source_info" : {
            "filename" : "action_chain1.p4",
            "line" : 63,
            "column" : 0,
            "source_fragment" : "table test1 { ..."
          },
          "key" : [
            {
              "match_type" : "ternary",
              "target" : ["data", "f1"],
              "mask" : null
            }
          ],
          "match_type" : "ternary",
          "type" : "simple",
          "max_size" : 1024,
          "with_counters" : false,
          "support_timeout" : false,
          "direct_meters" : null,
          "action_ids" : [17, 13, 4],
          "actions" : ["setb1", "noop", "NoAction"],
          "base_default_next" : "ex1",
          "next_tables" : {
            "setb1" : "ex1",
            "noop" : "ex1",
            "NoAction" : "ex1"
          },
          "default_entry" : {
            "action_id" : 4,
            "action_const" : false,
            "action_data" : [],
            "action_entry_const" : false
          }
        },
        {
          "name" : "ex1",
          "id" : 1,
          "source_info" : {
            "filename" : "action_chain1.p4",
            "line" : 70,
            "column" : 0,
            "source_fragment" : "table ex1 { ..."
          },
          "key" : [
            {
              "match_type" : "ternary",
              "target" : ["extra[0]", "h"],
              "mask" : null
            }
          ],
          "match_type" : "ternary",
          "type" : "simple",
          "max_size" : 1024,
          "with_counters" : false,
          "support_timeout" : false,
          "direct_meters" : null,
          "action_ids" : [5, 6, 7, 8, 9, 0],
          "actions" : ["set0b1", "act1", "act2", "act3", "noop", "NoAction"],
          "base_default_next" : null,
          "next_tables" : {
            "act1" : "tbl1",
            "act2" : "tbl2",
            "act3" : "tbl3",
            "set0b1" : null,
            "noop" : null,
            "NoAction" : null
          },
          "default_entry" : {
            "action_id" : 0,
            "action_const" : false,
            "action_data" : [],
            "action_entry_const" : false
          }
        },
        {
          "name" : "tbl1",
          "id" : 2,
          "source_info" : {
            "filename" : "action_chain1.p4",
            "line" : 80,
            "column" : 0,
            "source_fragment" : "table tbl1 { ..."
          },
          "key" : [
            {
              "match_type" : "ternary",
              "target" : ["data", "f2"],
              "mask" : null
            }
          ],
          "match_type" : "ternary",
          "type" : "simple",
          "max_size" : 1024,
          "with_counters" : false,
          "support_timeout" : false,
          "direct_meters" : null,
          "action_ids" : [14, 10, 1],
          "actions" : ["setb2", "noop", "NoAction"],
          "base_default_next" : null,
          "next_tables" : {
            "setb2" : null,
            "noop" : null,
            "NoAction" : null
          },
          "default_entry" : {
            "action_id" : 1,
            "action_const" : false,
            "action_data" : [],
            "action_entry_const" : false
          }
        },
        {
          "name" : "tbl2",
          "id" : 3,
          "source_info" : {
            "filename" : "action_chain1.p4",
            "line" : 83,
            "column" : 0,
            "source_fragment" : "table tbl2 { ..."
          },
          "key" : [
            {
              "match_type" : "ternary",
              "target" : ["data", "f2"],
              "mask" : null
            }
          ],
          "match_type" : "ternary",
          "type" : "simple",
          "max_size" : 1024,
          "with_counters" : false,
          "support_timeout" : false,
          "direct_meters" : null,
          "action_ids" : [15, 11, 2],
          "actions" : ["set1b1", "noop", "NoAction"],
          "base_default_next" : null,
          "next_tables" : {
            "set1b1" : null,
            "noop" : null,
            "NoAction" : null
          },
          "default_entry" : {
            "action_id" : 2,
            "action_const" : false,
            "action_data" : [],
            "action_entry_const" : false
          }
        },
        {
          "name" : "tbl3",
          "id" : 4,
          "source_info" : {
            "filename" : "action_chain1.p4",
            "line" : 86,
            "column" : 0,
            "source_fragment" : "table tbl3 { ..."
          },
          "key" : [
            {
              "match_type" : "ternary",
              "target" : ["data", "f2"],
              "mask" : null
            }
          ],
          "match_type" : "ternary",
          "type" : "simple",
          "max_size" : 1024,
          "with_counters" : false,
          "support_timeout" : false,
          "direct_meters" : null,
          "action_ids" : [16, 12, 3],
          "actions" : ["set2b2", "noop", "NoAction"],
          "base_default_next" : null,
          "next_tables" : {
            "set2b2" : null,
            "noop" : null,
            "NoAction" : null
          },
          "default_entry" : {
            "action_id" : 3,
            "action_const" : false,
            "action_data" : [],
            "action_entry_const" : false
          }
        }
      ],
      "action_profiles" : [],
      "conditionals" : []
    },
    {
      "name" : "egress",
      "id" : 1,
      "init_table" : null,
      "tables" : [],
      "action_profiles" : [],
      "conditionals" : []
    }
  ],
  "checksums" : [],
  "force_arith" : [],
  "extern_instances" : [],
  "field_aliases" : [
    [
      "queueing_metadata.enq_timestamp",
      ["standard_metadata", "enq_timestamp"]
    ],
    [
      "queueing_metadata.enq_qdepth",
      ["standard_metadata", "enq_qdepth"]
    ],
    [
      "queueing_metadata.deq_timedelta",
      ["standard_metadata", "deq_timedelta"]
    ],
    [
      "queueing_metadata.deq_qdepth",
      ["standard_metadata", "deq_qdepth"]
    ],
    [
      "intrinsic_metadata.ingress_global_timestamp",
      ["standard_metadata", "ingress_global_timestamp"]
    ],
    [
      "intrinsic_metadata.lf_field_list",
      ["standard_metadata", "lf_field_list"]
    ],
    [
      "intrinsic_metadata.mcast_grp",
      ["standard_metadata", "mcast_grp"]
    ],
    [
      "intrinsic_metadata.resubmit_flag",
      ["standard_metadata", "resubmit_flag"]
    ],
    [
      "intrinsic_metadata.egress_rid",
      ["standard_metadata", "egress_rid"]
    ]
  ]
}