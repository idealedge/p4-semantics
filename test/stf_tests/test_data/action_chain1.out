<T> <k> @nextPacket </k> <headers> <header> <name> %standard_metadata_t </name> <opts> .Map </opts> <fields> egress_spec : 32 ( .FieldMods ) ;  ( egress_spec : 32 ( .FieldMods ) ;  .FieldDecs ) </fields> </header> <header> <name> extra_t </name> <opts> "$fixed_width" |-> 32 </opts> <fields> h : 16 ( .FieldMods ) ;  ( b1 : 8 ( .FieldMods ) ;  ( b2 : 8 ( .FieldMods ) ;  .FieldDecs ) ) </fields> </header> <header> <name> data_t </name> <opts> "$fixed_width" |-> 96 </opts> <fields> f1 : 32 ( .FieldMods ) ;  ( f2 : 32 ( .FieldMods ) ;  ( h1 : 16 ( .FieldMods ) ;  ( b1 : 8 ( .FieldMods ) ;  ( b2 : 8 ( .FieldMods ) ;  .FieldDecs ) ) ) ) </fields> </header> </headers> <arrays> extra |-> $array ( 4 , 2 , 1 ) </arrays> <fieldLists> .FieldListCellBag </fieldLists> <fieldListCalcs> .FieldListCalcCellBag </fieldListCalcs> <calcFields> .CalcFieldCellBag </calcFields> <instances> <instance> <valid> true </valid> <metadata> true </metadata> <typeName> %standard_metadata_t </typeName> <name> standard_metadata </name> <fieldVals> ingress_port |-> @val ( 0 , 32 , false ) egress_spec |-> @val ( 2 , 32 , false ) </fieldVals> </instance> <instance> <valid> true </valid> <metadata> false </metadata> <typeName> extra_t </typeName> <name> $element ( extra , 0 ) </name> <fieldVals> h |-> @val ( 9509 , 16 , false ) b2 |-> @val ( 255 , 8 , false ) b1 |-> @val ( 39 , 8 , false ) </fieldVals> </instance> <instance> <valid> true </valid> <metadata> false </metadata> <typeName> extra_t </typeName> <name> $element ( extra , 1 ) </name> <fieldVals> h |-> @val ( 13107 , 16 , false ) b2 |-> @val ( 127 , 8 , false ) b1 |-> @val ( 40 , 8 , false ) </fieldVals> </instance> <instance> <valid> false </valid> <metadata> false </metadata> <typeName> extra_t </typeName> <name> $element ( extra , 2 ) </name> <fieldVals> .Map </fieldVals> </instance> <instance> <valid> false </valid> <metadata> false </metadata> <typeName> extra_t </typeName> <name> $element ( extra , 3 ) </name> <fieldVals> .Map </fieldVals> </instance> <instance> <valid> true </valid> <metadata> false </metadata> <typeName> data_t </typeName> <name> data </name> <fieldVals> h1 |-> @val ( 771 , 16 , false ) f2 |-> @val ( 33686018 , 32 , false ) f1 |-> @val ( 16843009 , 32 , false ) b2 |-> @val ( 102 , 8 , false ) b1 |-> @val ( 127 , 8 , false ) </fieldVals> </instance> </instances> <initInstances> <instances>-fragment <instance> <valid> true </valid> <metadata> true </metadata> <typeName> %standard_metadata_t </typeName> <name> standard_metadata </name> <fieldVals> ingress_port |-> @undef egress_spec |-> @undef </fieldVals> </instance> <instance> <valid> false </valid> <metadata> false </metadata> <typeName> extra_t </typeName> <name> $element ( extra , 0 ) </name> <fieldVals> .Map </fieldVals> </instance> <instance> <valid> false </valid> <metadata> false </metadata> <typeName> extra_t </typeName> <name> $element ( extra , 1 ) </name> <fieldVals> .Map </fieldVals> </instance> <instance> <valid> false </valid> <metadata> false </metadata> <typeName> extra_t </typeName> <name> $element ( extra , 2 ) </name> <fieldVals> .Map </fieldVals> </instance> <instance> <valid> false </valid> <metadata> false </metadata> <typeName> extra_t </typeName> <name> $element ( extra , 3 ) </name> <fieldVals> .Map </fieldVals> </instance> <instance> <valid> false </valid> <metadata> false </metadata> <typeName> data_t </typeName> <name> data </name> <fieldVals> .Map </fieldVals> </instance> </instances>-fragment </initInstances> <initArrays> extra |-> $array ( 4 , 0 , -1 ) </initArrays> <parserStates> <state> <name> extra </name> <body> ( extract ( extra [ next ] ) ; .ExtractOrSetStatements ) return select ( ( latest . b2 ) , .FieldOrDataRefs ) { ( ( 0x80 mask 0x80 ) , .ValueOrMaskeds ) : extra ;  ( default : ingress ;  .CaseEntries ) } </body> </state> <state> <name> start </name> <body> ( extract ( data ) ; .ExtractOrSetStatements ) return extra ; </body> </state> </parserStates> <valueSets> .ValueSetCellBag </valueSets> <exceptionHandlers> .ExceptionHandlerCellBag </exceptionHandlers> <handledExceptions> .Set </handledExceptions> <statefuls> .StatefulCellBag </statefuls> <tableDirects> .Map </tableDirects> <actions> <action> <name> act3 </name> <params> val , .ParamList </params> <body> modify_field ( ( extra [ 0 ] . b1 ) , ( val , .Args ) ) ;  .ActionStatements </body> </action> <action> <name> act2 </name> <params> val , .ParamList </params> <body> modify_field ( ( extra [ 0 ] . b1 ) , ( val , .Args ) ) ;  .ActionStatements </body> </action> <action> <name> act1 </name> <params> val , .ParamList </params> <body> modify_field ( ( extra [ 0 ] . b1 ) , ( val , .Args ) ) ;  .ActionStatements </body> </action> <action> <name> set2b2 </name> <params> val , .ParamList </params> <body> modify_field ( ( extra [ 2 ] . b2 ) , ( val , .Args ) ) ;  .ActionStatements </body> </action> <action> <name> set1b1 </name> <params> val , .ParamList </params> <body> modify_field ( ( extra [ 1 ] . b1 ) , ( val , .Args ) ) ;  .ActionStatements </body> </action> <action> <name> set0b1 </name> <params> val , .ParamList </params> <body> modify_field ( ( extra [ 0 ] . b1 ) , ( val , .Args ) ) ;  .ActionStatements </body> </action> <action> <name> setb2 </name> <params> val , .ParamList </params> <body> modify_field ( ( data . b2 ) , ( val , .Args ) ) ;  .ActionStatements </body> </action> <action> <name> noop </name> <params> .ParamList </params> <body> .ActionStatements </body> </action> <action> <name> setb1 </name> <params> port , ( val , .ParamList ) </params> <body> modify_field ( ( data . b1 ) , ( val , .Args ) ) ;  ( modify_field ( ( standard_metadata . egress_spec ) , ( port , .Args ) ) ;  .ActionStatements ) </body> </action> </actions> <actionProfiles> .ActionProfileCellBag </actionProfiles> <actionSelectors> .ActionSelectorCellBag </actionSelectors> <tables> <table> <name> tbl3 </name> <reads> ( data . f2 ) : ternary ;  .FieldMatchs </reads> <acts> actions { set2b2 ;  ( noop ;  .ActionNameItems ) } </acts> <opts> .TableOptionals </opts> <rules> .List </rules> <default> . </default> </table> <table> <name> tbl2 </name> <reads> ( data . f2 ) : ternary ;  .FieldMatchs </reads> <acts> actions { set1b1 ;  ( noop ;  .ActionNameItems ) } </acts> <opts> .TableOptionals </opts> <rules> ListItem ( $rule ( 5 , $ctr ( ListItem ( $pair ( @val ( 33685504 , 32 , false ) , @val ( 4294901760 , 32 , false ) ) ) ) , @call ( set1b1 , $resolved ( ListItem ( @val ( 40 , 0 , false ) ) ) ) ) ) </rules> <default> . </default> </table> <table> <name> tbl1 </name> <reads> ( data . f2 ) : ternary ;  .FieldMatchs </reads> <acts> actions { setb2 ;  ( noop ;  .ActionNameItems ) } </acts> <opts> .TableOptionals </opts> <rules> ListItem ( $rule ( 3 , $ctr ( ListItem ( $pair ( @val ( 33685504 , 32 , false ) , @val ( 4294901760 , 32 , false ) ) ) ) , @call ( setb2 , $resolved ( ListItem ( @val ( 38 , 0 , false ) ) ) ) ) ) </rules> <default> . </default> </table> <table> <name> ex1 </name> <reads> ( extra [ 0 ] . h ) : ternary ;  .FieldMatchs </reads> <acts> actions { set0b1 ;  ( act1 ;  ( act2 ;  ( act3 ;  ( noop ;  .ActionNameItems ) ) ) ) } </acts> <opts> .TableOptionals </opts> <rules> ListItem ( $rule ( 4 , $ctr ( ListItem ( $pair ( @val ( 9509 , 16 , false ) , @val ( 65535 , 16 , false ) ) ) ) , @call ( act2 , $resolved ( ListItem ( @val ( 39 , 0 , false ) ) ) ) ) ) ListItem ( $rule ( 2 , $ctr ( ListItem ( $pair ( @val ( 9472 , 16 , false ) , @val ( 65280 , 16 , false ) ) ) ) , @call ( act1 , $resolved ( ListItem ( @val ( 37 , 0 , false ) ) ) ) ) ) </rules> <default> . </default> </table> <table> <name> test1 </name> <reads> ( data . f1 ) : ternary ;  .FieldMatchs </reads> <acts> actions { setb1 ;  ( noop ;  .ActionNameItems ) } </acts> <opts> .TableOptionals </opts> <rules> ListItem ( $rule ( 1 , $ctr ( ListItem ( $pair ( @val ( 514 , 32 , false ) , @val ( 65535 , 32 , false ) ) ) ) , @call ( setb1 , $resolved ( ListItem ( @val ( 3 , 0 , false ) ) ListItem ( @val ( 7 , 0 , false ) ) ) ) ) ) ListItem ( $rule ( 0 , $ctr ( ListItem ( $pair ( @val ( 257 , 32 , false ) , @val ( 65535 , 32 , false ) ) ) ) , @call ( setb1 , $resolved ( ListItem ( @val ( 2 , 0 , false ) ) ListItem ( @val ( 127 , 0 , false ) ) ) ) ) ) </rules> <default> . </default> </table> </tables> <controlFlows> <control> <name> ingress </name> <body> apply ( test1 ) { .HitMissCases }  ( apply ( ex1 ) { ( act1 { apply ( tbl1 ) { .HitMissCases }  .ControlStatements } )  ( ( act2 { apply ( tbl2 ) { .HitMissCases }  .ControlStatements } )  ( ( act3 { apply ( tbl3 ) { .HitMissCases }  .ControlStatements } )  .ActionCases ) ) }  .ControlStatements ) </body> </control> </controlFlows> <cfset> SetItem ( ingress ) </cfset> <calcFieldSet> .Set </calcFieldSet> <ctx> <frameStack> .List </frameStack> <crnt> "$rule" |-> . "$table" |-> . "$action" |-> set1b1 "$miss" |-> false </crnt> </ctx> <packet> "000000010000000100000001000000010000001000000010000000100000001000000011000000110101010101100110001001010010010111111111111111110011001100110011111111110111111100000001000000100000001100000100" </packet> <packetout> .List </packetout> <parser> <graph> <onpath> .Set </onpath> <marked> <mstates> SetItem ( start ) SetItem ( extra ) </mstates> <mheaders> SetItem ( data ) SetItem ( extra ) </mheaders> </marked> <dporder> ListItem ( data ) ListItem ( extra ) </dporder> </graph> <pctx> <index> 160 </index> <lastExt> $element ( extra , 1 ) </lastExt> <varWidth> . </varWidth> <packetSize> @val ( 24 , 0 , false ) </packetSize> </pctx> </parser> <buffer> <in> .List </in> <out> ListItem ( $packet ( ListItem ( @val ( 257 , 32 , false ) ) ListItem ( @val ( 514 , 32 , false ) ) ListItem ( @val ( 771 , 16 , false ) ) ListItem ( @val ( 127 , 8 , false ) ) ListItem ( @val ( 102 , 8 , false ) ) ListItem ( @val ( 30583 , 16 , false ) ) ListItem ( @val ( 136 , 8 , false ) ) ListItem ( @val ( 0 , 8 , false ) ) , 2 ) ) ListItem ( $packet ( ListItem ( @val ( 514 , 32 , false ) ) ListItem ( @val ( 771 , 32 , false ) ) ListItem ( @val ( 1028 , 16 , false ) ) ListItem ( @val ( 7 , 8 , false ) ) ListItem ( @val ( 102 , 8 , false ) ) ListItem ( @val ( 30583 , 16 , false ) ) ListItem ( @val ( 136 , 8 , false ) ) ListItem ( @val ( 0 , 8 , false ) ) , 3 ) ) ListItem ( $packet ( ListItem ( @val ( 16843009 , 32 , false ) ) ListItem ( @val ( 33686018 , 32 , false ) ) ListItem ( @val ( 771 , 16 , false ) ) ListItem ( @val ( 127 , 8 , false ) ) ListItem ( @val ( 38 , 8 , false ) ) ListItem ( @val ( 9472 , 16 , false ) ) ListItem ( @val ( 37 , 8 , false ) ) ListItem ( @val ( 127 , 8 , false ) ) ListItem ( @val ( 16909060 , 32 , false ) ) , 2 ) ) ListItem ( $packet ( ListItem ( @val ( 16843009 , 32 , false ) ) ListItem ( @val ( 33686018 , 32 , false ) ) ListItem ( @val ( 771 , 16 , false ) ) ListItem ( @val ( 127 , 8 , false ) ) ListItem ( @val ( 102 , 8 , false ) ) ListItem ( @val ( 9509 , 16 , false ) ) ListItem ( @val ( 39 , 8 , false ) ) ListItem ( @val ( 255 , 8 , false ) ) ListItem ( @val ( 13107 , 16 , false ) ) ListItem ( @val ( 40 , 8 , false ) ) ListItem ( @val ( 127 , 8 , false ) ) ListItem ( @val ( 16909060 , 32 , false ) ) , 2 ) ) </out> </buffer> </T>
